src/transports/dash/segment_parser.ts

```diff
@@ -69,18 +76,43 @@ export default function parser({ content,
                                                             0),
                           appendWindow: [period.start, period.end] });
   } else { // it is an initialization segment
+    const { privateInfos } = segment;
+    const shouldExtractCompleteInitChunk = privateInfos !== undefined &&
+                                           privateInfos.shouldGuessInitRange === true;
+
+    const completeInitChunk = shouldExtractCompleteInitChunk ?
+      extractCompleteInitChunk(chunkData) : chunkData;
+
+    if (completeInitChunk === null &&
+        (
+          nextSegments === null ||
+          nextSegments.length === 0
+        )
+    ) {
+      if (!shouldExtractCompleteInitChunk) {
+        throw new Error("Can't extract complete init chunk and segment" +
+                        "references from loaded data.");
+      }
+      representation.index._addSegments([{ time: 0,
+                                           duration: Number.MAX_VALUE,
+                                           timescale: 1 }]);
+    }
+
     if (nextSegments !== null && nextSegments.length > 0) {
       representation.index._addSegments(nextSegments);
     }
 
-    const timescale = isWEBM ? getTimeCodeScale(chunkData, 0) :
-                               getMDHDTimescale(chunkData);
+    let timescale = null;
+    if (completeInitChunk !== null) {
+      timescale = isWEBM ? getTimeCodeScale(completeInitChunk, 0) :
+                           getMDHDTimescale(completeInitChunk);
+    }
 
     const chunkInfos = timescale != null && timescale > 0 ? { time: 0,
```

Comment:
I just checked, timescale cannot be `undefined` here

Reply:
Done